web/js/browser-support.js

```diff
@@ -0,0 +1,78 @@
+function webAssemblySupported() {
+	return "WebAssembly" in window
+		&& typeof window.WebAssembly.instantiate === "function";
+}
+
+function serviceWorkersSupported() {
+	return "serviceWorker" in navigator;
+}
+
+function webWorkersSupported() {
+	return "Worker" in window;
+}
+
+function indexedDbSupported() {
+	return "indexedDB" in window
+		&& typeof window.indexedDB.open === "function";
+}
+
+function sessionStorageSupported() {
+	var localStorageAvailable = "localStorage" in window
+		&& typeof window.localStorage.setItem === "function";
+
+	var sessionStorageAvailable = "sessionStorage" in window
+		&& typeof window.sessionStorage.setItem === "function";
+
+	return localStorageAvailable && sessionStorageAvailable;
+}
+
+function webCryptoSupported() {
+	return "crypto" in window
+		&& typeof window.crypto.randomUUID === "function";
+}
+
+function textEncodeDecodeSupported() {
+	return "TextEncoder" in window && "TextDecoder" in window;
+}
+
+function storeManagerPersistSupported() {
+	return "storage" in window.navigator
+		&& typeof window.navigator.storage.persist === "function";
+}
+
+function notificationApiSupported() {
+	return "Notification" in window;
+}
+
+function credentialsContainerSupported() {
+	return "credentials" in window.navigator
+		&& typeof window.navigator.credentials.get === "function";
+}
+
+function getUserMediaSupported() {
+	return "mediaDevices" in window.navigator
+		&& typeof window.navigator.mediaDevices.getUserMedia === "function";
+}
+
+function minimumFeaturesSupported() {
+	return (
+		webAssemblySupported()
+		&& serviceWorkersSupported()
+		&& webWorkersSupported()
+		&& indexedDbSupported()
+		&& sessionStorageSupported()
+		&& webCryptoSupported()
+		&& textEncodeDecodeSupported()
+		&& storeManagerPersistSupported()
+	);
+}
+
+function optionalFeaturesSupported() {
+	return (
+		notificationApiSupported()
+		&& credentialsContainerSupported()
+		&& getUserMediaSupported()
+	);
+}
+
+export { minimumFeaturesSupported, optionalFeaturesSupported };
```

Comment:
Should this module just auto-run these tests, and export the `true` / `false` results? I'm not seeing any reason to defer and require the including code to have to invoke the functions.

Reply:
I don't have a preference either way. If you want me to change that, I can, just let me know!

Reply:
I think for now, it's the simpler option. We can always change it back later if we need more sophistication.